pkgs/development/python-modules/nuitka/default.nix

```diff
@@ -1,44 +1,37 @@
-{ lib, stdenv
+{ lib
 , buildPythonPackage
+, ccache
 , fetchFromGitHub
-, vmprof
 , isPyPy
-, pkgs
-, scons
-, chrpath
+, ordered-set
+, python3
+, setuptools
 }:
 
 buildPythonPackage rec {
   pname = "nuitka";
-  version = "1.1.5";
-  format = "setuptools";
+  version = "1.8.4";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "Nuitka";
     repo = "Nuitka";
     rev = version;
-    hash = "sha256-8eWOcxATVS866nlN39b2VU1CuXAfcn0yQsDweHS2yDU=";
+    hash = "sha256-spa3V9KEjqmwnHSuxLLIu9hJk5PrRwNyOw72sfxBVKo=";
   };
 
-  nativeCheckInputs = [ vmprof ];
-  nativeBuildInputs = [ scons ];
-  propagatedBuildInputs = [ chrpath ];
+  nativeCheckInputs = [ ccache ordered-set ];
+  nativeBuildInputs = [ setuptools ];
```

Comment:
`ordered-set` should go to `propagatedBuildInputs` and we also need to add `zstandard`.

https://github.com/Nuitka/Nuitka/blob/5a2d18f027031a9b1b39e81ff3c03e08f0ca172b/setup.py#L360-L370

Reply:
i didn't think that they should go into the propagatedBuildInputs as they are just build time and not runtime dependencies and there is no propagatedNativeBuildInputs list so figured it is just up to the caller to add them to the native inputs, if needed.

Reply:
ok -- i was confused. ordered-set and zstandard are not showing up in the references of packages that use nuitka. change is updated with your suggestion.